docs/book/v3/storage/adapter-plugin-manager.md

```diff
@@ -0,0 +1,28 @@
+# Adapter Plugin Manager
+
+The `AdapterPluginManager` extends the laminas-servicemanager `AbstractPluginManager`, and has the following behaviors:
+
+- It will only return `Laminas\Cache\Storage\StorageInterface` instances.
+- All services are not shared by default; a new instance will be created each time you call `get()`.
```

Comment:
An introduction would be good here: Why should a user use this plug-in manager, what are the benefits and what are the advantages?

Example: https://github.com/laminas/laminas-form/pull/196/files#diff-3a6ef754f12eb3c6e8ccb63159f571bf326bf50fdf9304706dc9188e3af06f77R1-R20

Reply:
@froschdesign We do already have this plugin manager and this PR just adds another way to provide cache adapters which are not provided by laminas.

I personally do not see a reason why userland has a need for this plugin manager (besides the fact projects provide own cache adapters for niche backends). Those who already do provide a cache adapters do have to register these via a delegator factory and thus do already register these delegators to the `service_manager`/`dependencies` key so that their adapter is being added to the plugin manager. This PR just provides a dedicated cache key so that there is no runtime delegator to be executed on each request.

Not sure why we should provide a "why should a user use this plugin manager", esp. since userland should use `StorageAdapterFactoryInterface` instead which is already documented here: 
https://github.com/laminas/laminas-cache/blob/3.11.x/docs/book/v3/storage/adapter.md#quick-start

Does that suit our needs so far or do we have to add your example?

Reply:
@boesing 
I am not judging here whether the plugin manager is needed or not, but the newly added documentation. And if the plugin manager is to be described, then there must also be a justification of what advantages it offers and why the user should use it.

If you say that the plugin manager is not needed, then this documentation is unnecessary too.

Reply:
The plugin manager is needed once you create your own storage backend. and in this case, you can now - with this Patch - provide factories for your own written storage adapters in the same way as in every other component => via config. 

Thats all it provides, reading config and passing it to the plugin manager.

I dont see why we have to add more documentation than how to pass a valid config to the plugin manager factory. But maybe I miss your point?

Reply:
Let's approach it differently: Should the user act directly with the plugin manager at any point, e.g. pass it to a controller and then fetch an adapter from the plugin manager there?
If not, then we don't need detailed documentation, but if the user deals directly with the plugin manager - not only via the configuration - then we need a detailed documentation for it.

Reply:
The main suggestion by `laminas-cache` is to not use the plugin manager but to use `StorageAdapterFactoryInterface` instead, which is a dedicated service to create a storage adapter based on configuration.

We do use the https://github.com/laminas/laminas-cache/blob/3.11.x/src/Service/StorageCacheAbstractServiceFactory.php along with `caches` array config to fetch adapters from the config rather than using neither a service or the plugin manager.

Reply:
> The main suggestion by `laminas-cache` is to not use the plugin manager…

Then goes this current documentation in the wrong direction and we need something like "Configuration" under the section "Application Integration". Here can be described the new configuration key `storage_adapters`.
The plugin manager itself then only appears in a side sentence, with the reference that it is an implementation of the plugin manager based on laminas-servicemanager.